src/Runner.jl

```diff
@@ -802,6 +802,31 @@ function generate_compiler_wrappers!(platform::AbstractPlatform; bin_path::Abstr
     for tool in default_tools
         symlink("$(target)-$(tool)", joinpath(bin_path, triplet(platform), tool))
     end
+
+    # Generate other fake system-specific tools.
+    if os(platform) == "macos"
+        # `sw_vers -productVersion` is needed to make CMake correctly initialise the macOS
+        # platform, ref: <https://github.com/JuliaPackaging/Yggdrasil/pull/4403>.  In the
+        # future we may add more arguments.
+        sw_vers_path = joinpath(bin_path, triplet(platform), "sw_vers")
+        write(sw_vers_path, """
+              #!/bin/sh
+              if [[ "\${*}" == "-productVersion" ]]; then
+                  echo "$(macos_version(platform))"
+              fi
+              """)
+        chmod(sw_vers_path, 0o775)
+
+        # `xcrun` is another macOS-specific tool, which is occasionally needed to run some
+        # commands, for example for the CGO linker.  Ref:
+        # <https://github.com/JuliaPackaging/Yggdrasil/pull/2962>.
+        xcrun_path = joinpath(bin_path, triplet(platform), "xcrun")
```

Comment:
Since this isn't config dependent, should we just put this script in the shard?

Reply:
I'm not quite sure `xcrun` can always be emulated with just `exec "${@}"`, it worked in one case but I presume it's more complicated than that?  Having it here lets us quickly update it if/when necessary, without having to rebuild the full compiler shards, for all versions.

Reply:
I’d also like to use JIT’ed wrappers more and baked shards less, at least until it’s trivial to create and use such shards.